go/ct/evm_test.go

```diff
@@ -102,3 +105,246 @@ func TestCt_ExplicitCases(t *testing.T) {
 		})
 	}
 }
+
+const (
+	fuzzIdealStackSize     int = 7  // < max pops in a single instruction
+	fuzzMaximumCodeSegment int = 33 // < 1 instruction with 32 data bytes
+)
+
+// prepareInitialTestData is a helper function to be used by similar fuzzing tests
+// the arguments passed to the f.Add function needs to match the arguments
+// passed to the f.Fuzz function in type, position and number
+func prepareInitialTestData(f *testing.F, rnd *rand.Rand) {
+
+	for revision := R07_Istanbul; revision <= R13_Cancun; revision++ {
+		for op := 0x00; op < 0xFF; op++ {
+			for _, gas := range []int64{0, 1, 6, 10, 1000, math.MaxInt64} {
+
+				// generate a code segment with the operation followed by 6 random values
+				ops := [fuzzMaximumCodeSegment]byte{}
+				rand.Read(ops[:])
+				ops[0] = byte(op)
+
+				// generate a stack
+				stack := [fuzzIdealStackSize * 32]byte{}
+				// fill a quarter with random values
+				var i int
+				for i = 0; i < 7*8; i++ {
+					stack[i] = byte(rnd.Int31n(256))
+				}
+				// leave a quarter with zeros
+				for ; i < 7*16; i++ {
+					stack[i] = 0
+				}
+				// fill a quarter with max values
+				for ; i < 7*24; i++ {
+					stack[i] = 255
+				}
+				// fill a quarter with avg values
+				for ; i < 7*32; i++ {
+					stack[i] = 127
+				}
+
+				f.Add(
+					ops[:],         // opCodes
+					int64(gas),     // gas
+					byte(revision), // revision
+					stack[:],       // stack
+				)
+			}
+		}
+	}
+
+	// add one more with a full stack
+	ops := [fuzzMaximumCodeSegment]byte{}
+	rnd.Read(ops[:])
+	ops[0] = byte(0x00)
+	fullStack := [1024]byte{}
+	rnd.Read(fullStack[:])
+	f.Add(
+		ops[:],             // opCodes
+		int64(0),           // gas
+		byte(R07_Istanbul), // revision
+		fullStack[:],       // stack
+	)
+}
+
+// errorReportString is a helper function to print a summary of the test states.
+// Fuzzing generates a single error at the time, but when tunning the initial test
+// data, such states are evaluated in parallel and errors are produced in large numbers.
+// This function aims to present a compact overview of the most significant differences to
+// keep process output small and readable.
+func errorReportString(original *st.State, resultState *st.State, referenceState *st.State) string {
+	return fmt.Sprintf(`
```

Comment:
`st.State` has a [`Diff`](https://github.com/Fantom-foundation/Tosca/blob/c90d7bfdd232c87e22a1238ead1f9a46d3f2c6f4/go/ct/st/state.go#L363) function that would produce a list of differences between two states. Would this be helpful for this function?

Reply:
`st.State` also has a [String](https://github.com/Fantom-foundation/Tosca/blob/c90d7bfdd232c87e22a1238ead1f9a46d3f2c6f4/go/ct/st/state.go#L243) function that (should) produces human readable output.

Reply:
- State.String is not used to keep output small, when tuning a new fuzzer with an existing corpus errors are produced in large numbers, I find is useful to see the whole of them instead of the details. Which is something you can always get with a debug. Example: #549 
- State.Diff is used in the last comparison printout.